internal/core/data/router_test.go

```diff
@@ -69,8 +70,7 @@ var TestValueDescriptor3 = contract.ValueDescriptor{
 func TestRestValueDescriptorsUsageHandler(t *testing.T) {
 	Configuration = &ConfigurationStruct{Service: config.ServiceInfo{MaxResultCount: 10}}
 	defer func() { Configuration = &ConfigurationStruct{} }()
-	LoggingClient = logger.MockLogger{}
-	httpErrorHandler = errorconcept.NewErrorHandler(LoggingClient)
+	httpErrorHandler = errorconcept.NewErrorHandler(logger.NewMockClient())
```

Comment:
I think these should be moved inside the test loop.  

This ensures fresh instances for each test (and avoids the potential for cross-test contamination).

I would also ensure that you're using a single logging instance for the test; you're most recent set of changes injects a different logger mock into httpErrorHandler than the logger mock passed to restValueDescriptorsUsageHandler() (and I total understand why you did that given the current structure of the code).